numba/core/types/misc.py

```diff
@@ -500,6 +500,9 @@ def __init__(self, name):
     def iterator_type(self):
         return UnicodeIteratorType(self)
 
+    @property
```

Comment:
I am not sure this is the best way to solve this, since conceptually adding a `yield_type` to the `UnicodeType` object is strange. I would expect `yield_type` to be something associated with generators. Maybe it would make more sense to solve it like so:

https://github.com/numba/numba/pull/5294/files#diff-ae60f5afa43090234f80b19d5450963fR459-R477

But of course, since this code is needed in several different places, it probably makes the most sense to place it in common utilities module somewhere. We discussed this in a few developer meetings already and came to the conclusion, that  `numba.core.utils` might be a good home for it.

Reply:
Thanks for that, fixed it up.

And I guess after #5294 gets merged, we will have to refactor a lot of the implementations to use the `_guess_dtype` function.

Reply:
yeah, indeed there will be some refactoring later on.